### PR TITLE
Xml lists are extracted incorrectly

### PR DESCRIPTION
**Pull request by Schwusch**
This PR is addressing #245, where list members seem to be extracted incorrectly.

### CODE REVIEW

**Comment by Schwusch** [comment]
@isoos Does this seem correct to you? Especially the SNS/SES changes. 
It seems weird that it would've been wrong for so long. On the other hand, the result of the faulty extraction seems to have been empty lists that hasn't raised enough eyebrows until now.

File: generator/lib/builders/library_builder.dart [hunk 1]
```diff
@@ -420,7 +420,7 @@ String _xmlExtractorFn(
     return '_s.extractXml${_uppercaseName(dartType)}Value($elemVar, \'$elemName\')${enumeration ? '?.to${parent.className}()' : ''}';
   } else if (type == 'list') {
     final memberShape = api.shapes[shapeRef.member.shape];
-    final memberElemName = shapeRef.member.locationName ?? elemName;
+    final memberElemName = shapeRef.member.locationName ?? 'member';
```
Comment (xvrh) [inline generator/lib/builders/library_builder.dart:423]: From aws-js-sdk: https://github.com/aws/aws-sdk-js/blob/master/lib/xml/node_parser.js#L100
It doesn't use `locationName` but `name` (that you don't have right now in this implementation).

**Comment by isoos** [comment]
It would be nice to see an example response that is like this. Is there any documentation page that has it?

**Comment by Schwusch** [comment]
> It would be nice to see an example response that is like this

@xvrh eloquently provided one in #245, but yeah, documentation seems sparse

File: generator/lib/builders/library_builder.dart (hunk 1 above)
Comment (Schwusch) [reply generator/lib/builders/library_builder.dart:423]: I think it's a little bit more complicated than that, I think `shape.member` is also a `shape`, and the `name` property is defined [here](https://github.com/aws/aws-sdk-js/blob/master/lib/model/shape.js#L28).
Not sure whether we should apply the same name hierarchy here though. I'm not sure what [`memberName`](https://github.com/aws/aws-sdk-js/blob/master/lib/model/shape.js#L29) is at that point of execution.

**Comment by Schwusch** [comment]
I actually found a [PDF](https://docs.aws.amazon.com/IAM/latest/APIReference/iam-api.pdf) with an example response, page 280 :sweat_smile:

File: generator/lib/builders/library_builder.dart (hunk 1 above)
Comment (xvrh) [reply generator/lib/builders/library_builder.dart:423]: Oh ok. Pretty complicated yeah ;-)

**Comment by xvrh** [comment]
Similarly, the same problem exists for extracting `Map`. 
It should default to `key` and `value` names.

You can search for `_s.extractXmlStringValue(c, 'null')` in the generated apis, you will find lots of incorrect extractions.

**Comment by Schwusch** [comment]
I think I will mothball this change until #249 is in, so that we can have tests checking if we're doing the right thing.

**Comment by xvrh** [comment]
@Schwusch yes we should wait. The tests reveals that the change is not correct in the case of "flattened" list.
Also, I have a fix for the Map extraction.

**Comment by Schwusch** [comment]
> Also, I have a fix for the Map extraction.

We can close this PR if you have a more comprehensive fix, especially if it's based on test results :)

**Comment by Schwusch** [comment]
Abandon in favor of #265

**Pull request closed without merging**